metabase/metabase
# Extract sidebar component from DatabaseEditingApp

Preparation to solve #17450

As we investigate a bug in database creation, we learn that:

1. a single React component handles creating plus editing a database, procedurally
2. included in it is a sidebar that does operations to an existing database

In this PR we extract the sidebar into its own component and fully cover it with unit tests.

With this we want to make the sidebar quicker to understand, and also the database form file.

## How to Test

Edit your database, probably at: http://localhost:3000/admin/databases/1

Everything in the sidebar below should be usable:

<img src=https://user-images.githubusercontent.com/380816/131025653-17e02e4b-3ca1-4b4f-87c0-4209be4b8735.png width=300 />

These features are already covered by e2e tests as well: frontend/test/metabase/scenarios/admin/databases/edit.cy.spec.js
